core/src/main/java/org/elasticsearch/cluster/metadata/MetaData.java

```diff
@@ -915,6 +925,20 @@ public Builder customs(ImmutableOpenMap<String, Custom> customs) {
             return this;
         }
 
+        public Builder indexGraveyard(final IndexGraveyard indexGraveyard) {
+            putCustom(IndexGraveyard.TYPE, indexGraveyard);
+            return this;
+        }
+
+        public IndexGraveyard.Builder indexGraveyardBuilder() {
```

Comment:
we don't use this pattern in this class elsewhere. I think returning a builder is confusing as it may make people think that it changes the underlying graveyard, which it doesn't. Can we remove this?